### PR TITLE
Feature/stats2

### PR DESCRIPTION
**Pull request by kuenishi**
- Add Riak access stats, mochiweb stats and system stats
- Some code cleanup

### CODE REVIEW

File: src/stanchion_multipart.erl [hunk 1]
```diff
@@ -184,11 +184,17 @@
 
 check_no_multipart_uploads(Bucket, RiakPid) ->
     HashBucket = stanchion_utils:to_bucket_name(objects, Bucket),
-    {ok, Keys} = riakc_pb_socket:list_keys(RiakPid, HashBucket),
+
+
```
Comment (shino) [inline src/stanchion_multipart.erl]: Two blank lines.

**Comment by shino** [comment]
Will some tests come in the next round?

**Comment by kuenishi** [comment]
Those tests will be in riak_cs repo as integration test. Fair?

**Comment by shino** [comment]
Yup. Just want there be an issue or jira ticket for that (to cover my poor memory).

File: src/stanchion_manifest_fsm.erl [hunk 2]
```diff
@@ -1,346 +0,0 @@
-%% ---------------------------------------------------------------------
```
Comment (shino) [inline src/stanchion_manifest_fsm.erl:1]: Great eliminator :fire:

**Comment by shino** [comment]
The module `stanchion_wm_buckets` is used from velvet but not used _from riak cs_.
There is another PR for riak cs to merge velvet, so it seems good time to
delete the API completely.

Suggestion:
- Remove `stanchion_wm_buckets.erl`
- Then `stanchion_utils:get_buckets()` will be unused,
  `get_value()` in `stanchion_utils` will be unused.
  At last, `[riakc, get_manifest]` stats key can be removed in `get_value` function
  and stats definition in `stanchion_stats`.
- Remove `velvet:list_buckets()` in velvet merge PR to cs

**Comment by shino** [comment]
There was one wrong point in the above comment. `[riakc, get_manifest]` is used in other parts, can not be removed.

File: src/stanchion_utils.erl [hunk 3]
```diff
@@ -771,13 +717,18 @@ get_keys_and_values(BucketName) ->
     {ok, riakc_obj:riakc_obj()} | {error, notfound}.
 get_manifests_raw(RiakcPid, Bucket, Key) ->
     ManifestBucket = to_bucket_name(objects, Bucket),
-    riakc_pb_socket:get(RiakcPid, ManifestBucket, Key).
+    {Res, TAT} = ?TURNAROUND_TIME(riakc_pb_socket:get(RiakcPid, ManifestBucket, Key)),
+    stanchion_stats:update([riakc, get_manifests], TAT),
```
Comment (shino) [inline src/stanchion_utils.erl]: Existing r_t caches typo :joy_cat: 

Error log was as follows:

```
2015-07-23 11:41:42.343 [debug] <0.224.0>@stanchion_wm_bucket:delete_resource:123 Bucket: <<"riak-test-bucket">> Requester: <<"FSUNXVTYOMMTSX3Z_XJX">>
2015-07-23 11:41:42.345 [debug] <0.207.0>@stanchion_stats:update:95 Updating [riakc,get_cs_bucket] (1139)
2015-07-23 11:41:42.349 [debug] <0.207.0>@stanchion_stats:update:95 Updating [riakc,list_all_manifest_keys] (4026)
2015-07-23 11:41:42.350 [debug] <0.207.0>@stanchion_stats:update:95 Updating [riakc,get_manifests] (1005)
2015-07-23 11:41:42.350 [error] <0.207.0>@stanchion_utils:do_bucket_op:558 Error on updating bucket riak-test-bucket: {error,{badmatch,{error,not_found}}}
2015-07-23 11:41:42.350 [debug] <0.224.0>@stanchion_server:delete_bucket:105 [bucket,delete] {error,{error,{badmatch,{error,not_found}}}} 7580
2015-07-23 11:41:42.351 [debug] <0.224.0>@stanchion_stats:update:95 Updating [bucket,delete] (7580)
2015-07-23 11:41:42.358 [error] <0.224.0> Webmachine error at path "/buckets/riak-test-bucket" : {error,function_clause,[{stanchion_response,status_code,[{error,{badmatch,{error,not_found}}}],[{file,"src/stanchion_response.erl"},{line,65}]},{stanchion_response,api_error,3,[{file,"src/stanchion_response.erl"},{line,85}]},{webmachine_resource,resource_call,3,[{file,"src/webmachine_resource.erl"},{line,186}]},{webmachine_resource,do,3,[{file,"src/webmachine_resource.erl"},{line,142}]},{webmachine_decision_core,resource_call,1,[{file,"src/webmachine_decision_core.erl"},{line,48}]},{webmachine_decision_core,...},...]}
```

**Comment by shino** [comment]
Finished first round of review. Overall diff seems nice, bunch of code deletion is great :trophy: 
Will run r_t after comments are addressed.

**Comment by kuenishi** [comment]
Updated. Will run r_t with cs#1197, too.

**Comment by kuenishi** [comment]
Woops, updated

**Comment by kuenishi** [comment]
Fmm, many riak_test are failing with 500 ... cannot be merged yet.

**Comment by kuenishi** [comment]
Yay!

```
0 Tests Failed
34 Tests Passed
That's 100.0% for those keeping score

real    101m38.407s
user    15m54.576s
sys     3m42.104s
```

**Comment by shino** [comment]
:muscle:

**Pull request merged**